### PR TITLE
W-14447114 - added Graal.js to engines

### PR DESCRIPTION
**Pull request by JuanPabloFi**
I added Graal as a compile dependency. 
Since Nashorn is not included anymore after Java 15 then , when Nashorn is null it uses Graal for scripting. 
I added tests to cover Nashorn, Graal and Ecma.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -109,6 +109,16 @@
             <version>${commons.lang.version}</version>
             <scope>test</scope>
         </dependency>
+        <dependency>
+            <groupId>org.graalvm.js</groupId>
+            <artifactId>js</artifactId>
+            <version>22.0.0</version>
```
Comment (acurci-at-mulesoft) [inline pom.xml:115]: Please, replace the version number with a property.

Comment (juanfloresMule) [inline pom.xml:115]: we should use the properties for the dependencies version

**Review by acurci-at-mulesoft — CHANGES_REQUESTED** [review]

File: pom.xml (hunk 1 above)
Comment (JuanPabloFi) [reply pom.xml:115]: right, thanks

**Review by juanfloresMule — APPROVED** [review]
LGTM

**Review by acurci-at-mulesoft — APPROVED** [review]

**Pull request merged**